src/main/java/com/jagrosh/discordipc/IPCClient.java

```diff
@@ -180,127 +289,186 @@ public final ClientBuild getClientBuild()
     {
         ACTIVITY_JOIN,
         ACTIVITY_SPECTATE,
-        ACTIVITY_JOIN_REQUEST;
+        ACTIVITY_JOIN_REQUEST,
+        /**
+         * A backup key, only important if the
+         * IPCClient receives an unknown event
+         * type in a JSON payload.
+         */
+        UNKNOWN;
         
-        public static Subscription of(String str)
+        static Subscription of(String str)
         {
-            for(Subscription s: Subscription.values())
-                if(s.name().equalsIgnoreCase(str))
+            for(Subscription s : Subscription.values())
+            {
+                if(s != UNKNOWN && s.name().equalsIgnoreCase(str))
                     return s;
-            return null;
+            }
+            return UNKNOWN;
         }
     }
-    
-    
+
+
     // Private methods
     
     /**
      * Makes sure that the client is connected (or not) depending on if it should
-     * for the current state
-     * @param connected 
+     * for the current state.
+     *
+     * @param connected Whether to check in the context of the IPCClient being
+     *                  connected or not.
      */
     private void checkConnected(boolean connected)
     {
         if(connected && status != Status.CONNECTED)
-            throw new IllegalStateException("Not connected!");
+            throw new IllegalStateException(String.format("IPCClient (ID: %d) is not connected!", clientId));
         if(!connected && status == Status.CONNECTED)
-            throw new IllegalStateException("Already connected!");
+            throw new IllegalStateException(String.format("IPCClient (ID: %d) is already connected!", clientId));
     }
     
     /**
-     * Starts a reading thread, which automatically ends when disconnected or closed
+     * Initializes this IPCClient's {@link IPCClient#readThread readThread}
+     * and calls the first {@link #read()}.
      */
     private void startReading()
     {
-        readThread = new Thread(() ->
-        {
+        readThread = new Thread(() -> {
             try
             {
                 Packet p;
                 while((p = read()).getOp() != OpCode.CLOSE)
                 {
-                    if(listener!=null && p.getJson().has("cmd") && p.getJson().getString("cmd").equals("DISPATCH"))
+                    JSONObject json = p.getJson();
+                    if(listener != null && json.has("cmd") && json.getString("cmd").equals("DISPATCH"))
                     {
-                        switch(Subscription.of(p.getJson().getString("evt")))
+                        JSONObject data = json.getJSONObject("data");
+
+                        switch(Subscription.of(json.getString("evt")))
                         {
                             case ACTIVITY_JOIN:
-                                listener.onActivityJoin(this, p.getJson().getJSONObject("data").getString("secret"));
+                                LOGGER.debug("Reading thread received a 'join' command.");
```

Comment:
Change this and similar log statements from 'command' to 'event' as per the official docs